oak-core/src/main/java/org/apache/jackrabbit/oak/query/ast/FullTextSearchImpl.java

```diff
@@ -144,10 +144,9 @@ public FullTextExpression getFullTextConstraint(SelectorImpl s) {
             // e.g. ?{"inferenceModelConfig": "ada-test-model"}?little red fox
             // So here we split the query into text part of query and inferenceConfig part of query.
             // Afterwards we only parse text part of query as this part of query is what we want to search.
-            if (query.getSettings().isInferenceEnabled()) {
-                VectorQuery vectorQuery = new VectorQuery(rawText);
-                queryText = vectorQuery.getQueryText();
-            }
+            // We are explicitly removing vectorQueryConfig from raw query
```

Comment:
Comment messages are not consistent. In the LucenePropertyIndex, similar code has comment 

`Lucene indexes don't support inference, so we should remove queryInferenceConfig`

Reply:
FulltextSearchImpl is executed prior to specific implementation of lucene or elastic. There we are removing the queryVectorConfig completely. Here are only providing query part to be parsed by FullTextParser but later send full query as  raw text.